### ISSUE TITLE
Failed to get nested archive for entry BOOT-INF/lib/moneta-1.4.2.pom after upgrading to 1.2.0

### ISSUE THREAD

**Issue by winniewithana** [zalando/jackson-datatype-money#260]
## Description
I am working on a spring boot project and upgrading the jackson-datatype-money to 1.2.0 which causes the jar file is failed to unpacked because the pom file is included in the jar for this version.

## Expected Behavior
It should work as same as 1.1.0 which is not included the pom file.

## Actual Behavior
I think it's probably why it ends up including the pom file
https://github.com/zalando/jackson-datatype-money/blob/main/pom.xml#L86
then causes the issue that I have here:
```
Exception in thread "main" java.lang.IllegalStateException: Failed to get nested archive for entry BOOT-INF/lib/moneta-1.4.2.pom
        at org.springframework.boot.loader.archive.JarFileArchive.getNestedArchive(JarFileArchive.java:108)
        at org.springframework.boot.loader.archive.JarFileArchive.getNestedArchives(JarFileArchive.java:86)
        at org.springframework.boot.loader.ExecutableArchiveLauncher.getClassPathArchives(ExecutableArchiveLauncher.java:70)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:49)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: java.io.IOException: Unable to open nested jar file 'BOOT-INF/lib/moneta-1.4.2.pom'
        at org.springframework.boot.loader.jar.JarFile.getNestedJarFile(JarFile.java:256)
        at org.springframework.boot.loader.jar.JarFile.getNestedJarFile(JarFile.java:241)
        at org.springframework.boot.loader.archive.JarFileArchive.getNestedArchive(JarFileArchive.java:103)
        ... 4 more
Caused by: java.io.IOException: Unable to find ZIP central directory records after reading 17305 bytes
        at org.springframework.boot.loader.jar.CentralDirectoryEndRecord.<init>(CentralDirectoryEndRecord.java:65)
        at org.springframework.boot.loader.jar.CentralDirectoryParser.parse(CentralDirectoryParser.java:52)
        at org.springframework.boot.loader.jar.JarFile.<init>(JarFile.java:123)
        at org.springframework.boot.loader.jar.JarFile.<init>(JarFile.java:111)
        at org.springframework.boot.loader.jar.JarFile.createJarFileFromFileEntry(JarFile.java:289)
        at org.springframework.boot.loader.jar.JarFile.createJarFileFromEntry(JarFile.java:264)
        at org.springframework.boot.loader.jar.JarFile.getNestedJarFile(JarFile.java:252)
```

## Steps to Reproduce
1. upgrade the jackson-datatype-money from 1.1.0 to 1.2.0

## Your Environment

* Version used: Spring Boot 2.1.3.RELEASE.

**Comment by whiskeysierra** [comment]
That's really weird. I found this old commit which introduced that: https://github.com/zalando/jackson-datatype-money/commit/9b396e1a45a101445dafd788938d3dc1ee019bc9

But I can't remember why. I guess I didn't realize back then that the groupId of moneta changed from `org.javamoney` to `org.javamoney.moneta` and I didn't find the right dependency anymore. This `pom` dependency still pulls in the correct ones transitively, but there is no reason for a such a weird construct.

Good catch! I'll prepare a fix.

**Comment by whiskeysierra** [comment]
I just released 1.2.1 with a bugfix to maven central. Should be available soon. Please let me know if it doesn't solve your issue.

Thanks for reporting this!
